server/src/main/java/io/crate/role/Role.java

```diff
@@ -65,17 +69,23 @@ public static Properties fromXContent(XContentParser parser) throws IOException
                     case "secure_hash":
                         secureHash = SecureHash.fromXContent(parser);
                         break;
+                    case "jwt":
+                        jwtProperties = JwtProperties.fromXContent(parser);
+                        break;
                     default:
                         throw new ElasticsearchParseException(
                             "failed to parse role properties, unexpected field name: " + parser.currentName()
                         );
                 }
             }
-            return new Properties(login, secureHash);
+            return new Properties(login, secureHash, jwtProperties);
         }
 
         public Properties(StreamInput in) throws IOException {
-            this(in.readBoolean(), in.readOptionalWriteable(SecureHash::readFrom));
+            this(in.readBoolean(),
+                 in.readOptionalWriteable(SecureHash::readFrom),
+                 in.getVersion().onOrAfter(Version.V_5_7_0) ? in.readOptionalWriteable(JwtProperties::readFrom) : null
```

Comment:
here we use `null`, but elsewhere (requests) we use empty maps, maybe we should be consistent?

Reply:
We use nullable record. I think there was a confusing fro the old implementation where Map was used, I removed it in https://github.com/crate/crate/pull/15640/commits/5bd489a2379ac72a9effbea9d881e1f5dc685268#diff-05ff3ea28f321d32053d4ba4043a7006962031aed40b321d33df54ba72620317L88

Reply:
:+1: